### ISSUE TITLE
xcms.merger: raise an error when a sample isn't in the sampleMetadata

### ISSUE THREAD

**Issue by lecorguille** [workflow4metabolomics/xcms#57]
An example:

- The  sampleMetadata:
```
sampleMetadata	class	polarity	injOrder	sampleType	batch
20170209_P_Blanc12	blank	positive	38	blank	1
20170209_P_Blanc13	blank	positive	52	blank	1
20170209_P_QC01	pool	positive	11	pool	1
20170209_P_QC02	pool	positive	25	pool	1
20170418_P_S01n01	TV_nd	positive	58	sample	1
20170418_P_S01n02	TV_nd	positive	69	sample	1
```

- The xcms.merger logs
```
	XSET OBJECT INFO
                    class
20170418_P_Blanc12   <NA>
20170418_P_Blanc13   <NA>
20170418_P_QC01      <NA>
20170418_P_QC02      <NA>
20170418_P_S01n01   TV_nd
20170418_P_S01n02   TV_nd
```

So the tool should check and raise and error if it met this use-case.

**Comment by lecorguille** [comment]
- [ ] Fix for XCMS 1.48.0: https://github.com/workflow4metabolomics/xcms/pull/79
- [ ] Fix for XCMS 3.0.0: https://github.com/workflow4metabolomics/xcms/pull/78/commits/4c42ad29fda7c144ba95580f2799b513b8396bd1#diff-00e99a457a9b83549c7dd8f26e57ec77R58

- [ ] Push to the Testtoolshed and the Toolshed

- [ ] Update in https://galaxy.workflow4metabolomics.org/

**Comment by lecorguille** [comment]
ping @bernt-matthias
Thanks for your report
